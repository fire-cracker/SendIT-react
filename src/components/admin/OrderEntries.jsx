import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';


const OrderEntries = ({ showPending, orders }) => {
  const order = orders.map((item, index) => {
    const {
      fromAddress, toAddress, weight, price, orderStatus, createdAt, presentLocation
    } = item;

    const date = format(
      new Date(createdAt),
      'MM/DD/YYYY, hh:mm a'
    );
    return (
      <tr key={parseInt(index.toString(), 7)}>
        <td>{fromAddress}</td>
        <td>{toAddress}</td>
        <td>{weight}</td>
        <td>
          &#8358;
          {price}
        </td>
        <td>{date}</td>
        <td>{orderStatus}</td>
        {showPending ? null : (<td>{presentLocation}</td>)}
      </tr>
    );
  });

  return (
    order
  );
};

OrderEntries.propTypes = {
  orders: PropTypes.array,
};

export default OrderEntries;
