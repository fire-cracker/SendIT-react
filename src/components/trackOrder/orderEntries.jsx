import React from 'react';
import PropTypes from 'prop-types';
// import { format } from 'date-fns';


const OrderEntries = ({ orders }) => {
  const order = orders.map((item, index) => {
    const {
      fromAddress, toAddress, weight, price, orderStatus, createdAt
    } = item;

    return (
      <tr key={parseInt(index.toString(), 7)}>
        <td>{fromAddress}</td>
        <td>{toAddress}</td>
        <td>{weight}</td>
        <td>
          &#8358;
          {price}
        </td>
        <td>{createdAt}</td>
        <td>{orderStatus}</td>
      </tr>
    );
  });

  return (
    order
  );
};

OrderEntries.propTypes = {
  orders: PropTypes.array,
};

export default OrderEntries;
