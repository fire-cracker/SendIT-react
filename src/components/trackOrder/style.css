/* Styling the body background */
body {
  margin: 0px;
  width: 100%;
  height: 100%;
  padding: 0px;
  font-style: normal;
  line-height: 1.5;
  color: #232323;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1rem;
  font-weight: 400;
}
.track--preview>h1{
  text-align: center;
  margin: auto;
  width: 95%;
  height: max-content;
  color: black;
  font-size: 2rem;
  border-radius: 5px;
  margin-bottom: 5px;
  font-weight: bold;
  padding: 10px;
  font-family: 'Rubik', sans-serif;
}

/* Styling the Parcel delivery Orders Section background */
#information{
  height: max-content;
  height: 100%;
  width: 100%;
}

/* Styling the Table */
table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

td, th {
  border: 1px solid #dddddd;
  text-align: left;
  padding: 8px;
}

tr:nth-child(even) {
  background-color: #dddddd;
}

/* Styling the Top background Section */
#section3 {
  height: 170px;
  width: 100%;
  /* background-image: url("img/background3.jpg"); */
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}
/* Styling the Top nav Section */
.navTabs {
  display: -webkit-flex;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  flex-wrap: wrap;
  border-bottom: none;
  padding-top: 1rem!important;
  margin-top: 3rem!important;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}
.navLink {
  box-sizing: border-box;
  color: #767676;
  font-style: normal;
  font-weight: 500;
  padding: 1rem 3rem;
  margin: 0.4rem .8rem !important;
  transition: color .3s;
  border: 2px solid #767676;
  border-radius: 100px !important;
  font-family: 'Rubik', sans-serif;
  display: block;
}
a{
  text-decoration: none;
  cursor: pointer;
  background-color: transparent;
}
.navLink:hover, .navTabs>.active{
  border: 2px solid #149dcc;
  background: #149dcc;
  color:white;
}
.delivered,.newPending{
  display:none;
}
.newPending-active,.Delivered-active{
  display:block;
}

/* Styling the Parcel delivery Orders Section */
#informationFields{
margin-top: 20px;
height: max-content;
text-align: left;
}
.track--preview{
  margin: 10% 10% 0% 5%;
  height: 90%;
  width: 90%;
  background-color: rgba(51, 51, 51, 0);
  text-align: left;  
}
#shipmentOrderPreview{
margin: 2% 0% 0% 1%;
overflow: auto;
top: 0px;
width: 90%;	
height: 240px;
padding: 0px 4% 0px 4%;
border-bottom: 1px solid black;
border-left: 1px solid black;
border-right: 1px solid black;	
}
#createShipmentPreview{
  display:flex;
  justify-content: space-between;
  height: 240px;
}
.red {
  color: red;
  font-size: 1.5rem;
}
.edit__button{
  font-size: 1rem;
}
.no__order{
  text-align: center;
  font-size: 3rem;
}
/* For desktop: */
@media only screen and (min-width: 1200px) {
.track--preview{
      margin: 5% 10% 0% 5%;
}
h1{
  font-size: 200%;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
}
#shipmentOrderPreview {
  height: 190px;
}
}
@media only screen and (min-width: 600px) and (max-width: 1199px)  {

}