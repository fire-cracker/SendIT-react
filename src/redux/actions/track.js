import axios from '../../utils/axiosConfig';
import { GET_ORDERS_REQUEST, GET_ORDERS_SUCCESS } from './actionTypes';

const api = process.env.API_ROOT_URL;

export const getOrderRequest = () => ({
  type: GET_ORDERS_REQUEST
});

export const getOrderSuccess = data => ({
  type: GET_ORDERS_SUCCESS,
  orders: data
});

export const getOrder = userId => async (dispatch) => {
  try {
    dispatch(getOrderRequest());
    const { data } = await axios.get(`${api}/users/${userId}/parcels`);
    dispatch(getOrderSuccess(data));
  } catch (error) {
    throw error;
  }
};
