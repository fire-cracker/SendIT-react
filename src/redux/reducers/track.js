import {
  GET_ORDERS_REQUEST,
  GET_ORDERS_SUCCESS,
} from '../actions/actionTypes';


const orderInitialState = {
  orders: []
};
export default (state = orderInitialState, action) => {
  switch (action.type) {
    case GET_ORDERS_REQUEST:
      return {
        ...state
      };

    case GET_ORDERS_SUCCESS: {
      const { orders: { order } } = action;
      return {
        ...state,
        orders: order,
      };
    }
    default:
      return state;
  }
};
