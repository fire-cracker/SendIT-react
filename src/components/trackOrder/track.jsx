import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import decodeJwt from 'jwt-decode';

import { getOrder, cancelOrder, editOrderDestination } from '../../redux/actions/track';
import OrderEntries from './orderEntries';
import Header from '../header';
import './style.css';

export class GetUserOrders extends Component {
  state = {
    showNew: false,
    showPending: true,
    showDelivered: true,
    showInput: true,
    status: 'New'
  }

  componentDidMount() {
    const { token } = localStorage;
    const { userId } = decodeJwt(token);
    const { getOrder, history } = this.props;
    getOrder(userId)
      .then()
      .catch((error) => {
        const { response, response: { status } } = error;
        if (response && status === 401) {
          history.push('/');
          return toast.error('Your session has expired. You need to login');
        }
        toast.error('Unknown error');
      });
  }

  componentDidUpdate() {
    const { history, login: { isLoggedIn } } = this.props;
    if (isLoggedIn === false) {
      history.push('/');
    }
  }

  onClickCancel = (parcelId) => {
    const { history } = this.props;
    this.props.cancelOrder(parcelId)
      .then()
      .catch((error) => {
        const { response, response: { status } } = error;
        if (response && status === 401) {
          history.push('/');
          return toast.error('Your session has expired. You need to login');
        }
        toast.error('Unknown error');
      });
  }

  onClickEdit = (orderValue) => {
    this.setState(prevState => ({
      showInput: !prevState.showInput,
      destination: orderValue
    }));
  }

  onChangeDestination = (e) => {
    this.setState({
      destination: e.target.value
    });
  }

  onClickSubmit = (parcelId) => {
    const { history, editOrderDestination } = this.props;
    const { destination } = this.state;
    editOrderDestination(parcelId, destination)
      .then()
      .catch((error) => {
        const { response, response: { status } } = error;
        if (response && status === 401) {
          history.push('/');
          return toast.error('Your session has expired. You need to login');
        }
        toast.error('Unknown error');
      });
    this.onClickEdit();
  }

  toggleDiv = (status, showingDiv) => {
    this.setState({
      showNew: true,
      showPending: true,
      showDelivered: true,
      [showingDiv]: false,
      status
    });
  };


  render() {
    const { orders: { orders } } = this.props;
    const {
      showNew, showPending, showDelivered, showInput, status
    } = this.state;

    const orderEntries = orders.filter(order => order.orderStatus === status);
    const cancelSymbol = showPending && showNew;
    return (
      <div className="track">
        <div>
          <Header />
        </div>
        <div id="informationFields">
          <div className="track--preview">
            <h1> Hi User! Track Your Orders </h1>

            <div className="navTabs" id="navTabs">
              <a
                className={showNew ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('New', 'showNew')}
              >
                NEW ORDERS
              </a>
              <a
                className={showPending ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('Pending', 'showPending')}
              >
                PENDING ORDERS
              </a>
              <a
                className={showDelivered ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('Cancelled', 'showDelivered')}
              >
                CANCELLED ORDERS&nbsp;
              </a>
            </div>

            <div id="information">

              <div
                id="shipmentOrder"
                className="newPending-active"
              >
                { orderEntries.length
                  ? (
                    <div className="order-table">
                      <table>
                        <thead>
                          <tr>
                            <th>Pickup Location</th>
                            <th>Destination</th>
                            <th>weight(kg)</th>
                            <th>Price</th>
                            <th>Date of Order</th>
                            <th>Status</th>
                            {showPending ? null : (<th>PresentLocation</th>)}
                            {cancelSymbol ? null : (<th>Cancel</th>)}
                            {cancelSymbol ? null : (<th>Change Destination</th>)}
                          </tr>

                          <OrderEntries
                            cancelSymbol={cancelSymbol}
                            showInput={showInput}
                            showPending={showPending}
                            orders={orderEntries}
                            onClickCancel={parcelId => this.onClickCancel(parcelId)}
                            onClickEdit={this.onClickEdit}
                            onClickSubmit={parcelId => this.onClickSubmit(parcelId)}
                            onChangeDestination={this.onChangeDestination}
                          />

                        </thead>
                      </table>
                    </div>
                  )
                  : (
                    <p className="no__order">No Order</p>
                  )
                }

              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

GetUserOrders.propTypes = {
  getOrder: PropTypes.func,
  cancelOrder: PropTypes.func,
  editOrderDestination: PropTypes.func,
  orders: PropTypes.object,
  history: PropTypes.object,
  login: PropTypes.object
};
const mapStateToProps = ({ login, orders }) => ({ login, orders });

const mapDispatchToProps = {
  getOrder,
  cancelOrder,
  editOrderDestination
};

export default connect(mapStateToProps, mapDispatchToProps)(GetUserOrders);
