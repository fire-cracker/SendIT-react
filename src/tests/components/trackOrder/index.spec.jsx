import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import axios from '../../../utils/axiosConfig';

import { GetUserOrders } from '../../../components/trackOrder/track';
import { orders } from '../../mock/orders';
import OrderEntries from '../../../components/trackOrder/orderEntries';

jest.mock('../../../utils/axiosConfig');
jest.mock('react-toastify');
jest.mock('../../../redux/actions/createOrder');

const preventDefault = jest.fn();
let wrapper;

describe('Tests for the track Page', () => {
  const props = {
    getOrder: jest.fn().mockResolvedValue(true),
    orders: { orders }
  };
  let wrapper;
  beforeEach(() => {
    jest.spyOn(GetUserOrders.prototype, 'componentDidMount').mockImplementationOnce(() => true);
    wrapper = mount(<BrowserRouter><GetUserOrders {...props} /></BrowserRouter>);
  });
  it('Should render the track page', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('.navLink').at(0).simulate('click');
    wrapper.find('.navLink').at(1).simulate('click');
    wrapper.find('.navLink').at(2).simulate('click');
  });

  it('should get order if request is  correct', () => {
    const mocked = jest.spyOn(GetUserOrders.prototype, 'componentDidMount');
    expect(mocked.mock.calls.length).toBe(2);
  });
});
