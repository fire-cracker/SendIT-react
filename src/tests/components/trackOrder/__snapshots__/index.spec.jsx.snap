// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the track Page Should render the track page 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "5h1gs8",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "5h1gs8",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <GetUserOrders
      getOrder={
        [MockFunction] {
          "calls": Array [
            Array [
              "chinchin",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Promise {},
            },
          ],
        }
      }
      login={
        Object {
          "isLoggedIn": true,
        }
      }
      orders={
        Object {
          "orders": Array [
            Object {
              "fromAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "toAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "weight": "2",
            },
            Object {
              "fromAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "toAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "weight": "2",
            },
          ],
        }
      }
    >
      <div
        className="track"
      >
        <div>
          <Header>
            <div
              className="header"
              id="header"
            >
              <ToastContainer
                autoClose={5000}
                bodyClassName={null}
                className={null}
                closeButton={
                  <CloseButton
                    ariaLabel="close"
                  />
                }
                closeOnClick={true}
                draggable={true}
                draggablePercent={80}
                hideProgressBar={false}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-right"
                progressClassName={null}
                progressStyle={null}
                rtl={false}
                style={null}
                toastClassName={null}
                transition={[MockFunction]}
              />
              <header
                className="navBar"
                id="navBar"
              >
                <div
                  id="headerContent"
                >
                  <div
                    className="navigation"
                    id="navigation"
                  >
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="logout"
                      id="logout"
                      to="/"
                    >
                      <Route
                        path="\\\\/"
                      >
                        <Link
                          aria-current="page"
                          className="logout active"
                          id="logout"
                          replace={false}
                          style={Object {}}
                          to="/"
                        >
                          <a
                            aria-current="page"
                            className="logout active"
                            href="/"
                            id="logout"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            Logout
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="prefile"
                      id="profile"
                      to="/profile"
                    >
                      <Route
                        path="\\\\/profile"
                      >
                        <Link
                          aria-current={null}
                          className="prefile"
                          id="profile"
                          replace={false}
                          to="/profile"
                        >
                          <a
                            aria-current={null}
                            className="prefile"
                            href="/profile"
                            id="profile"
                            onClick={[Function]}
                          >
                            Profile
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="create__order"
                      id="create__order"
                      to="/createOrder"
                    >
                      <Route
                        path="\\\\/createOrder"
                      >
                        <Link
                          aria-current={null}
                          className="create__order"
                          id="create__order"
                          replace={false}
                          to="/createOrder"
                        >
                          <a
                            aria-current={null}
                            className="create__order"
                            href="/createOrder"
                            id="create__order"
                            onClick={[Function]}
                          >
                            Create Order
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                    <NavLink
                      activeClassName="active"
                      aria-current="page"
                      className="track"
                      id="track"
                      to="/track"
                    >
                      <Route
                        path="\\\\/track"
                      >
                        <Link
                          aria-current={null}
                          className="track"
                          id="track"
                          replace={false}
                          to="/track"
                        >
                          <a
                            aria-current={null}
                            className="track"
                            href="/track"
                            id="track"
                            onClick={[Function]}
                          >
                            Track
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                    <a
                      className="icon"
                      id="icon"
                      onClick={[Function]}
                    >
                      <i
                        className="fa fa-bars"
                      />
                    </a>
                  </div>
                </div>
                <div
                  className="blink_caption"
                  href="home.html"
                >
                  <div
                    className="blink"
                    id="blin"
                  >
                    Just SendIT!!!
                  </div>
                </div>
              </header>
            </div>
          </Header>
        </div>
        <div
          id="informationFields"
        >
          <div
            className="track--preview"
          >
            <h1>
               Hi User! Track Your Orders 
            </h1>
            <div
              className="navTabs"
              id="navTabs"
            >
              <a
                className="navLink active"
                onClick={[Function]}
              >
                NEW ORDERS
              </a>
              <a
                className="navLink"
                onClick={[Function]}
              >
                PENDING ORDERS
              </a>
              <a
                className="navLink"
                onClick={[Function]}
              >
                CANCELLED ORDERSÂ 
              </a>
            </div>
            <div
              id="information"
            >
              <div
                className="newPending-active"
                id="shipmentOrder"
              >
                <p
                  className="no__order"
                >
                  No Order
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </GetUserOrders>
  </Router>
</MemoryRouter>
`;
